paymentsheet/src/test/java/com/stripe/android/paymentsheet/ViewActionRecorder.kt

```diff
@@ -0,0 +1,19 @@
+package com.stripe.android.paymentsheet
+
+import com.google.common.truth.Truth.assertThat
+
+internal class ViewActionRecorder<VA> {
+    private val _viewActions: MutableList<VA> = mutableListOf()
+    val viewActions: List<VA>
+        get() = _viewActions.toList()
+
+    fun record(viewAction: VA) {
+        _viewActions += viewAction
+    }
+
+    fun consume(viewAction: VA) {
+        assertThat(_viewActions.size).isGreaterThan(0)
+        assertThat(_viewActions[0]).isEqualTo(viewAction)
+        _viewActions.removeAt(0)
```

Comment:
Nit: Should `_viewActions` be a queue instead?

```suggestion
        assertThat(_viewActions.remove()).isEqualTo(viewAction)
```

Reply:
I prefer the code that exists today, it makes it clear where the failures are.

Reply:
> Nit: Should _viewActions be a queue instead?

I think either would work fine!